react/ShelfContent.js

```diff
@@ -84,17 +85,18 @@ class ShelfContent extends Component {
   }
 
   ssrFallback() {
-    const { products, itemsPerPage } = this.props
-    const className = this.getClassByItemsPerPage(itemsPerPage)
+    const { products, itemsPerPage, isMobile } = this.props
+    const items = isMobile ? DEFAULT_ITEMS_MOBILE : itemsPerPage
+    const className = this.getClassByItemsPerPage(items)
     return (
       <div className="flex justify-center">
         {products &&
-          products.slice(0, itemsPerPage).map(item => {
+          products.slice(0, items).map((item, index) => {
```

Comment:
This `return` statement is not needed, you can simply do `[...].map((item, index) => ( <div>...</div> )`